pyo3-derive-backend/src/pyfunction.rs

```diff
@@ -0,0 +1,261 @@
+    fn add_literal(&mut self, item: &NestedMeta, lit: &syn::Lit) -> syn::Result<()> {
+        match lit {
+            syn::Lit::Str(ref lits) => {
+                // "*"
+                if lits.value() == "*" {
+                    if self.has_kwargs {
+                        return Err(syn::Error::new_spanned(
+                            item,
+                            "syntax error, keyword self.arguments is defined",
+                        ));
+                    }
+                    if self.has_varargs {
+                        return Err(syn::Error::new_spanned(
+                            item,
+                            "self.arguments already define * (var args)",
+                        ));
+                    }
+                    self.has_varargs = true;
+                    self.arguments.push(Argument::VarArgsSeparator);
+                } else {
+                    return Err(syn::Error::new_spanned(lits, "Unknown string literal"));
+                }
+            }
+            _ => {
+                return Err(syn::Error::new_spanned(
+                    item,
+                    format!("Only string literal is supported, got: {:?}", lit),
+                ));
+            }
+        };
+        Ok(())
+    }
+
+    fn add_work(&mut self, item: &NestedMeta, ident: &Ident) -> syn::Result<()> {
+        // self.arguments in form somename
+        if self.has_kwargs {
+            return Err(syn::Error::new_spanned(
+                item,
+                "syntax error, keyword self.arguments is defined",
+            ));
+        }
+        if self.has_kw {
+            return Err(syn::Error::new_spanned(
+                item,
+                "syntax error, argument is not allowed after keyword argument",
+            ));
+        }
+        self.arguments.push(Argument::Arg(ident.clone(), None));
+        Ok(())
+    }
+
+    fn add_name_value(&mut self, item: &NestedMeta, nv: &syn::MetaNameValue) -> syn::Result<()> {
+        match nv.lit {
+            syn::Lit::Str(ref litstr) => {
+                if litstr.value() == "*" {
+                    // args="*"
+                    if self.has_kwargs {
+                        return Err(syn::Error::new_spanned(
+                            item,
+                            "* - syntax error, keyword self.arguments is defined",
+                        ));
+                    }
+                    if self.has_varargs {
+                        return Err(syn::Error::new_spanned(item, "*(var args) is defined"));
+                    }
+                    self.has_varargs = true;
+                    self.arguments.push(Argument::VarArgs(nv.ident.clone()));
+                } else if litstr.value() == "**" {
+                    // kwargs="**"
+                    if self.has_kwargs {
+                        return Err(syn::Error::new_spanned(
+                            item,
+                            "self.arguments already define ** (kw args)",
```

Comment:
I don't understand why you mention `self.arguments` here.
Do users really need to know about this?